server/src/main/java/com/epam/aidial/core/server/controller/ApplicationController.java

```diff
@@ -60,29 +60,22 @@ public Future<?> getApplication(String applicationId) {
 
     public Future<?> getApplications() {
         Config config = context.getConfig();
-        List<ApplicationData> list = new ArrayList<>();
-
-        for (Application application : config.getApplications().values()) {
-            if (application.hasAccess(context.getUserRoles())) {
-                ApplicationData data = ApplicationUtil.mapApplication(application);
-                list.add(data);
+        Proxy proxy = context.getProxy();
+
+        return proxy.getVertx().executeBlocking(() -> {
+            List<Application> list = new ArrayList<>();
```

Comment:
Please don't run the code is not supposed to be blocking in worker threads.
The blocking op should be run in the block if the flag `isIncludeCustomApps` is set to true.

Reply:
The code uses heavy method com.epam.aidial.core.server.util.ApplicationTypeSchemaUtils#filterCustomClientProperties and must be run in a worker thread.